db/version_set.cc

```diff
@@ -797,6 +798,20 @@ Status VersionSet::LogAndApply(VersionEdit* edit, port::Mutex* mu) {
   }
   Finalize(v);
 
+  if (descriptor_log_ != nullptr) {
+    if (manifest_file_size_hint_ >= options_->write_buffer_size) {
+      Log(options_->info_log, "MANIFEST size %ld exceed, switch to next file\n",
+          long(manifest_file_size_hint_));
+      delete descriptor_log_;
+      delete descriptor_file_;
+      descriptor_log_ = nullptr;
+      descriptor_file_ = nullptr;
+      manifest_file_number_ = manifest_file_number_ + 1;
```

Comment:
Call NewFileNumber() here?

Reply:
You should set manifest_file_number_ after SetCurrentFile() successes. Otherwise, the DeleteObsoleteFiles() may delete the current valid manifest file.

Reply:
This can not happen because DeleteObsoleteFiles and LogAndApply both run in the background compact thread.